MarvelApp/MarvelApp/Network/Endpoints/HomePageEndpoints.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  HomePageEndpoints.swift
+//  MarvelApp
+//
+//  Created by Caner Tüysüz on 5.12.2024.
+//
+
+import Foundation
+
+enum HomepageEndpoints: EndpointProtocol {
+    case getCharacters
+
+    var url: String {
+        switch self {
+        case .getCharacters:
+            return "https://gateway.marvel.com:443/v1/public/characters"
+        }
+    }
+    
+    var method: HTTPMethod {
+        switch self {
+        case .getCharacters:
+            return .GET
+        }
+    }
+    
+    var headers: [String : String]? {
+        switch self {
+        case .getCharacters:
+            return ["Content-Type": "application/json"]
+        }
+    }
+    
+    var limit: Int {
+        switch self {
+        case .getCharacters:
+            return 20
+        }
+    }
+    
+    var queryItems: [URLQueryItem]? {
+        switch self {
+        case .getCharacters:
+            return [
```

Comment:
You can get query items as key/value, and finally map them as URLQueryItem in the network layer.

The reason for this;

What changes is the key and value. The class that holds this is URLQueryItem. This can only change when network dependencies change. If this is created in a common place, it will be simpler and more manageable.

```suggestion
            var queryItems: [String: Any]? {
         switch self {
         case .getCharacters:
             return [
                "key": Value
             ] ....   
```